src/action/transaction.rs

```diff
@@ -0,0 +1,240 @@
+use std::time::Duration;
+
+use crate::{
+    client::options::TransactionOptions,
+    options::{ReadConcern, WriteConcern},
+    selection_criteria::SelectionCriteria,
+    ClientSession,
+};
+
+use super::option_setters;
+
+impl ClientSession {
+    /// Starts a new transaction on this session. If no options are set, the session's
+    /// `defaultTransactionOptions` will be used. This session must be passed into each operation
+    /// within the transaction; otherwise, the operation will be executed outside of the
+    /// transaction.
+    ///
+    /// Errors returned from operations executed within a transaction may include a
+    /// [`crate::error::TRANSIENT_TRANSACTION_ERROR`] label. This label indicates that the entire
+    /// transaction can be retried with a reasonable expectation that it will succeed.
+    ///
+    /// Transactions are supported on MongoDB 4.0+. The Rust driver currently only supports
+    /// transactions on replica sets.
```

Comment:
```suggestion
    /// Transactions on replica sets are supported on MongoDB 4.0+. Transactions on sharded
    /// clusters are supported on MongoDB 4.2+.
```

(this was probably just a copy-paste but that info is outdated)